### DISCUSSION TITLE
How to link to other repos in readme.md files

### DISCUSSION THREAD

**Discussion by jezmck** [community/community#22070] [New to GitHub]
<p>I’d like to link to (e.g.) <a href="https://github.com/vuejs/vue" rel="nofollow noopener">https://github.com/vuejs/vue</a> in my personal readme, but it feels wrong to have to include the <code>https://github.com/</code> part.</p>
<p>Is there no way to have a link relative to the whole github site?</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi <a class="mention" href="/jezmck">@jezmck</a>!</p>
<p>I’m afraid we don’t support relative links across the site, no, sorry!</p>
<p>You can always use conventional markdown syntax to create links that are easier on the eye, though?</p>
<p>For example:</p>
<pre><code class="lang-auto">[Vue](https://github.com/vuejs/vue)
</code></pre>
<p>…is rendered as  <a href="https://github.com/vuejs/vue">Vue</a></p>
<p>Does that help?</p>

**Comment by jezmck** [comment]
<p>Okay. How do I turn this into a feature request?</p>

**Comment by yamiacat** [comment]
<p>You can submit a feature request <a href="https://support.github.com/contact/feedback">through our official product feedback form</a> so that our product team can see exactly how you’d like GitHub to work!</p>

**Comment by Dayan94** [comment]
I also want the same feature to be included in GitHub. Four years have passed and no change in the markdown file full URL writing in the same account's different repositories' readme files.

**Comment by jimmy-zhening-luo** [comment]
@jezmck, was this ever submitted as a feature request? I'm curious why GitHub can autolink humans like @jimmy-zhening-luo; organizations like @w3c; and issues, PRs, and even commits (by SHA!) inside of repos — but not the root of those same repos?

# These take me where I want to go ✅

- jlord/sheetsee.js#26  `repo -> issue`
- @jlord  `user root`
- @community  `org root`

# But these do not ❌

1. @community/community `repo root`
1. community/community `repo root`
1. jlord/sheetsee.js# `repo root`
1. jlord/sheetsee.js@ `repo root`

# Proposal

Of the negative example syntaxes, only (3) or (4) or some variation thereof seems like reasonable candidates.

- (1) is already reserved for tagging a team inside an organization
- (2) would false positive far too often

On the other hand, (3) and (4) could reasonably autolink IMO if the following are fulfilled:

- Standalone token (e.g. not part of URL or email address)
- When using `#`, possibly require another trigger word like `foo/bar#README`[^1] to take you to the root, or a second hash like `foo/bar##`.
- When using `@`, to avoid conflation with commits, IMO it is safest simply to require an *empth* `@`, e.g. `foo/bar@`. (For all you Microsofties, did you know that this is how Google engineers write aliases, which they call LDAPs? Everything really *is* opposite there.)
- Both cases maximally avoid symbol/namespace conflicts because the trigger tokens reside at the end of resolvable syntaxes.
- Both cases mitigate false positives with other `@` and `#` bounded items by requiring the link to be a standalone (bounded) token. GitHub also checks if the URL is resolvable as a precondition for rendering an autolink. Combined, this seems at least as safe from FPs as the existing autolinks supported by GitHub.

[^1]: N.B. The `#` syntax also makes relative linking possible which is what @Dayan94 requested, but I will note that what @jezmck requested was *not* relative linking, since they were simply asking to be able to reference a repo's *root*. Ironically, GitHub actually *does* support relative *auto*-linking like the aforementioned jlord/sheetsee.js#26, and relative *non*-auto-linking like [vuejs/vue](./../../../vuejs/vue) `[vuejs/vue](./../../../vuejs/vue)` 😂.

**Comment by jimmy-zhening-luo** [comment]
@yamiacat , I tried to open a feature request but the link you provided is dead, it just redirects to this same community discussion forum, in which there is no special `Feature Request` thread type that is distinct from this exsiting thread.

**Comment by h3rb** [comment]
bump, i also want this feature

**Comment by nexushoratio** [comment]
While looking for this feature, I discovered that, with enough `/../../...`, a Markdown link will work:

> [!NOTE]
It is hidden in the Images section, but appears to work for links as well:

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#images

But, as noted in https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#relative-links
> Relative links are easier for users who clone your repository. Absolute links may not work in clones of your repository - we recommend using relative links to refer to other files within your repository.

So, if someone clones the repo to a local disk, those absolute links are going to be useless.  Which makes sense, since that is a feature of the GH renderer, not Markdown syntax.

Anyway, in my experiments:

To a repo from a different org:
> \[Vue](/../../../../vuejs/vue)

[Vue](/../../../../vuejs/vue)

To a repo from the same org (obviously this one will not work):
> [Vue](/../../../vue)

[Vue](/../../../vue)

Though, since this is the community section, this will work to link to an org....
> [VueJS](/../../../vuejs)

[VueJS](/../../../vuejs)

I guess, just like from the command line, one has to carefully count the number of `..` used.

---

Still, consider if a project was hosted on a different service, how would this be implemented?  Would you expect each service to have its own way of linking?  GH:vuejs/vue  ?   Should there be a shared list of such prefixes that all services could use?  Again, what about non-source control uses of Markdown?

There is an old saying in computer science:
> All the world is not a VAX.

Plan for portability.  In the world in Markdown, that means using \[Markdown](https://www.example.com/path/to/resource).
